app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,205 @@
+        """Test retrieving a list of recipes starting with param
+        substring name"""
+        substr = 'Baked'
+        sample_recipe(name=f'{substr} Spaghetti')
+        sample_recipe(name=f'{substr} Mushroom')
+        sample_recipe(name=f'{substr} Chciken')
+        sample_recipe(name=f'Does not start with {substr}')
+
+        url = recipe_url('list')
+        res = self.client.get(url, {
+            'ordering': 'id',
+            'name': substr
+        })
+
+        recipes = Recipe.objects \
+            .filter(name__startswith=substr) \
+            .order_by('id')
+        serializer = RecipeSerializer(recipes, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_detail_recipe(self):
+        """Test retrieving a detailed recipe"""
+        recipe = sample_recipe()
+
+        url = recipe_url('detail', [recipe.id])
+        res = self.client.get(url)
+
+        serializer = RecipeSerializer(recipe)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_detail_undefined_recipe(self):
+        """Test retrieving a non existing recipe"""
+        url = recipe_url('detail', ['undefinedId'])
+        res = self.client.get(url)
+
+        self.assertEqual(res.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_create_recipe(self):
+        """Test creating a simple recipe"""
+        data = {
+            'name': 'Test recipe 1',
+            'description': 'Test recipe 1 description',
+            'ingredients': []
+        }
+
+        url = recipe_url('list')
+        content_type = 'application/json'
+        res = self.client.post(url, data, content_type=content_type)
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            pick(res.data, ['name', 'description']),
+            pick(data, ['name', 'description'])
+        )
+
+    def test_create_recipe_with_ingredients(self):
+        """Test creating a recipe with ingredients"""
+        data = {
+            'name': 'Test recipe 1',
+            'description': 'Test recipe 1 description',
+            'ingredients': [
+                {'name': 'Test ingredient 1'},
+                {'name': 'Test ingredient 2'},
+            ]
+        }
+
+        url = recipe_url('list')
+        content_type = 'application/json'
+        res = self.client.post(url, data, content_type=content_type)
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            pick(res.data, ['name', 'description']),
+            pick(data, ['name', 'description'])
+        )
+
+        ingredients = [pick(ingredient, ['name'])
+                        for ingredient in res.data['ingredients']]
+
+        self.assertEqual(len(ingredients), len(data['ingredients']))
+        self.assertEqual(ingredients, data['ingredients'])
```

Comment:
You might also want to check that there was a recipe created in the DB with those ingredients.